src/manager/crossmsgs.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+
+use std::str::FromStr;
+
+use anyhow::Result;
+use cid::Cid;
+use fil_actors_runtime::cbor;
+use fvm_shared::address::Address;
+use fvm_shared::clock::ChainEpoch;
+use fvm_shared::MethodNum;
+use ipc_sdk::subnet_id::SubnetID;
+
+use crate::constants::GATEWAY_ACTOR_ADDRESS;
+use crate::jsonrpc::JsonRpcClient;
+use crate::lotus::client::LotusJsonRPCClient;
+use crate::lotus::message::mpool::MpoolPushMessage;
+use crate::lotus::LotusClient;
+
+// Prototype function for submitting topdown messages. This function is supposed to be called each
+// Nth epoch of a parent subnet. It reads the topdown messages from the parent subnet and submits
+// them to the child subnet.
+async fn submit_topdown_msgs<T: JsonRpcClient + Send + Sync>(
+    parent_epoch: ChainEpoch,
+    account: &Address,
+    child_subnet: SubnetID,
+    child_client: &LotusJsonRPCClient<T>,
```

Comment:
It would be great to have an abstraction between the manager and Lotus, to work with your own domain types, and not assume that every child subnet implements Lotus' API.